proposals/dictionary-expressions.md

```diff
@@ -54,9 +61,44 @@ Choices here would have implications regarding potential syntactic ambiguities,
 
 ## Design Intuition
 
-Intuitively, *dictionary expressions* work similarly to *collection expressions*, except treating a `k:v` element as a shorthand for creating a `System.Collections.Generic.KeyValuePair<TKey, TValue>`.  Many rules for *dictionary expressions* will correspond to existing rules for *collection expressions*, just requiring aspects such as *element* and *iteration types* to be some `KeyValuePair<,>`.
+There are two core aspects to the design of dictionary expressions. 
+
+First is the concept of a `dictionary type`. *Dictionary types* are types that are similar to the existing *collection types*, with the additional requirements that they have an *element type* of some `KeyValuePair<TKey, TValue>` *and* have an indexer `TValue this[TKey] { ... }`.  The latter requirement ensures that `List<KeyValuePair<int, string>>` is not considered a dictionary type (as its indexer is from `int` to `KeyValuePair<int, string>`), while `Dictionary<TKey, TValue>` would of course be.
 
-*Dictionary types* are types that derive from `IEnumerable<KeyValuePair<TKey, TValue>>` and have an indexer `TValue this[TKey] { ... }`.  The latter requirement ensures that `List<KeyValuePair<int, string>>` is not considered a dictionary type (as its indexer is from `int` to `KeyValuePair<int, string>`).
+Second is that collection expressions containing `KeyValuePair<,>`s (coming from `expression_element`s, `spread_element`s, or `key_value_pair_element`s) can now instantiate a normal `collection type` *or* a `dictionary type`.  
```

Comment:
i also wanted to make it clear up front that `k:v` is not restrictedd to dictionary types, nor is it required to make a dictionary type.  It's just a really convenient shorthand that we will expect to be *primarily* used with dictionary types.  but it is both optional to use there, and can still be used for normal collections.